### PR TITLE
Python3

### PR DESCRIPTION
**Pull request by profxj**
First major attempt to port the code to Python 3.
Most of everything is cosmetic and should be
backwards compatible to Python2 (this is the intent).

I did have to introduce one "kludge" in alfunc_gaussian.py
when running on a model from Tiffany.  See line 97
and please inform me if this is correct.    If I don't
set getid to something here, I crash on line 103 (see the try/except).

Am getting the same results as Tiffany to within ~1e-3 precision.
Plots look good too..

### CODE REVIEW

File: alis/alfunc_base.py [hunk 1]
```diff
@@ -380,7 +381,7 @@ def parout(self, params, mp, istart, level, errs=None, reletter=False, conv=None
         Nothing should be changed here when writing a new function.
         --------------------------------------------------------
         """
-        if errs == None: errors = params
+        if np.all(errs == None): errors = params
```
Comment (rcooke-ast) [inline alis/alfunc_base.py]: I think this should probably be:

`if errs is None: errors = params`

File: alis/alfunc_gaussian.py [hunk 2]
```diff
@@ -94,12 +94,16 @@ def set_vars(self, p, level, mp, ival, wvrng=[0.0,0.0], spid='None', levid=None,
                         if mp['mlnk'][j][0] == mp['mtie'][ival][i]:
                             cmd = 'lnkprm = ' + mp['mlnk'][j][1]
                             exec(cmd)
+                getid = mp['tpar'][mp['mtie'][ival][i]][1]  # THIS IS A JXP KLUDGE.  WHAT IS CORRECT??
                 levadd += 1
             else:
                 getid = level+levadd
                 levadd+=1
             if lnkprm is None:
-                params[i] = self.parin(i, p[getid], parb)
```
Comment (rcooke-ast) [inline alis/alfunc_gaussian.py]: Is this still needed?

File: alis/alfunc_polynomial.py [hunk 3]
```diff
@@ -402,8 +402,10 @@ def parout(self, params, mp, istart, level, errs=None, reletter=False, conv=None
         Nothing should be changed here when writing a new function.
         --------------------------------------------------------
         """
-        if errs == None: errors = params
```
Comment (rcooke-ast) [inline alis/alfunc_polynomial.py]: As before, I think this should be:

`if errs is None: errors = params`

File: alis/alload.py [hunk 4]
```diff
@@ -477,7 +477,9 @@ def load_data(slf, datlines, data=None):
             filename = "Data read from 'data' keyword"
             linspl.insert(0,linspl[0])
         if not os.path.exists(filename) and data is None:
-            if not slf._argflag['generate']['data'] and not slf._isonefits: msgs.error("File does not exist -"+msgs.newline()+filename)
+            if not slf._argflag['generate']['data'] and not slf._isonefits:
+                import pdb; pdb.set_trace()
```
Comment (rcooke-ast) [inline alis/alload.py]: Should probably raise a warning here first?

File: alis/alfunc_gaussian.py [hunk 5]
```diff
@@ -94,12 +94,16 @@ def set_vars(self, p, level, mp, ival, wvrng=[0.0,0.0], spid='None', levid=None,
                         if mp['mlnk'][j][0] == mp['mtie'][ival][i]:
                             cmd = 'lnkprm = ' + mp['mlnk'][j][1]
                             exec(cmd)
+                getid = mp['tpar'][mp['mtie'][ival][i]][1]  # THIS IS A JXP KLUDGE.  WHAT IS CORRECT??
```
Comment (rcooke-ast) [inline alis/alfunc_gaussian.py]: I don't think this line is needed. Going through the various cases, `lnkprm=None` before the `if` loop (on line 85).

(1) The `if` case (line 87): `getid` is set and `lnkprm` is unchanged from None.

(2) The `elif` case (line 89): Further split into an `if/else` case; in both of these, lnkprm is set (on line 91, and then the combination of line 95+96). The only possibility where this might break is if the `if` statement on line 94 is never satisfied in the for loop, which would point to a bug somewhere else in the code.

(3) The `else` case (line 99): `getid` is set, and `lnkprm` is unchanged

Then, on line 102, we only need one of `lnkprm` and `getid` to be set. And this is (should!) be satisfied above.

I think the most likely problem is that the links are not defined correctly in the input model settings file, or the link being attempted is not permitted in the current version of ALIS. If you have an example model and data file, I can probably diagnose the problem a little better (or maybe `exec` is not allowed under python3?)

The bottom line is that this addition to the code shouldn't be here (the bug/problem is due to something else), but thanks for raising the issue!

**Review by rcooke-ast — CHANGES_REQUESTED** [review]
Thanks very much for your help in making ALIS python3 compatible. Just a few minor things to tweak - thanks again!

**Review by rcooke-ast — APPROVED** [review]
